### DISCUSSION TITLE
part-ii/2021-78/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#404] [Question comments]
Part II 2021 Graph Theory: Paper 3, Section II, 17G 

https://questions.tripos.org/part-ii/2021-78/

**Comment by walrus1729** [comment]
<details>
<summary>Solution</summary>

a) The first part is bookwork. The second part follows by induction on $n$, the base case $n=2$ being clear. If we remove a vertex $x$ , we will have a monochromatic spanning tree in the remaining $n-1$ vertices by the inductive hypothesis (without loss of generality, it is red). If $x$ has any red neighbours, then adding this edge in gives a red spanning tree on $n$ vertices, and if all of the neighbours of $x$ are blue, then we get a blue spanning tree by taking all of these edges. 

The claim is false for $3$ colours; for a counterexample, take a $K_3$ and give each edge a different colour.

b) This formula works by counting the number of paths of length $2$ with $x$ as a middle vertex (then summing over all vertices $x$). We have $d(x)$ choices for a starting vertex, and then our end vertex can be any neighbour of $x$ other than the starting vertex, giving $d(x) - 1$ possibilities. 

For the last part, we count the number of monochromatic paths of length $2$ in $2$ different ways; it is maybe slightly easier to count paths without direction. One way is to use the previous formula twice (for red paths and for blue paths) and then sum; recall that ${d(x) \choose 2} = \frac{1}{2}d(x)(d(x)-1)$. 

The other way is to consider the triangles in $G$. If we have $T$ monochromatic triangles, then each of these corresponds to $3$ monochromatic paths of length $2$ (ignoring direction). The total number of triangles in $G$ is ${n \choose 3}$, and the remaining triangles correspond to $1$ monochromatic path of length $2$ each (removing the edge whose colour only appears once.

Equating these two expressions, we get:

$$
\sum_{x} [{d_{r}(x) \choose 2} + {d_{b}(x) \choose 2}] = 3T + ({n \choose 3} - T)
$$

and rearranging gives the desired result.

</details>
